RD-5758 Migrate app framework - `maintenance` components

leon-good-life opened the pull request:
<!--- Provide JIRA issue ID and a general summary of your changes in the Title above -->
<!--- e.g. "RD-1234 Improve styling in SuperComponent" -->
<!--- Note that the title is used as a commit message after merging the PR -->

## Description
Jira ticket: RD-5758
<!--- Describe your changes to help reviewers understand the details. -->

## Screenshots / Videos
N/A
<!--- If applicable, add screenshot(s) or video(s) describing the changes you made. -->
<!--- Consider adding comparison screenshot(s) - before and after the change. -->
<!--- If not applicable, just write “N/A” in this section. -->

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I followed the [UI Testing Guidelines](https://cloudifysource.atlassian.net/l/cp/udCVfvrx) and verified that all tests/checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

all specs 11-DEC-2022:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4083/pipeline

all specs 8-DEC-2022 18:25 IL time:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4070/pipeline

all specs 8-DEC-2022:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4055/pipeline

all specs: ✅ all tests that pass on master branch passes on this branch too:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/3773/pipeline

<!--- If applicable, describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
<!--- If applicable, provide a link to system tests, e.g: -->
<!--- [#1039](https://jenkins.cloudify.co/.../1039) (2021-08-06 07:05:50) --> 
<!--- If not applicable, just write “N/A” in this section. -->

## Documentation
N/A
<!--- If applicable, describe how you documented or plan to document your changes. -->
<!--- Check UI documentation guidelines - https://cloudifysource.atlassian.net/l/c/EYWJ0XfB - for details. -->
<!--- Provide links to JIRA issues, other PRs or related documentation pages. -->
<!--- If not applicable, just write “N/A” in this section. -->

kubama commented:
I can still see `@ts-nocheck` in the file that is in scope of the ticket

leon-good-life commented on app/containers/maintenance/MaintenanceModePageMessage.ts:
```diff
@@ -1,13 +1,14 @@
-// @ts-nocheck File not migrated fully to TS
```
In this PR, I migrated to TypeScript. I need to also do here the ongoing refactor of moving code from `containers` to `components`. I suggest to do it in a separate PR within this ticket. Because this PR is already has many changes.

leon-good-life replied:
I converted MaintenanceModePageMessage to function component: bdbcd6615b7af7a37a3b3e7fa0dd819abcd36425
And I moved `MaintenanceModePageMessage` file from `containers` to `components`: b8a6ce0944442f16bb0ddc6ca44af4294b5dbf2a

leon-good-life commented on app/components/shared/MaintenanceModeModal.tsx:
```diff
@@ -224,11 +242,11 @@ function MaintenanceModeModal({
                                                         <Menu.Item
                                                             content={tExecutions('killCancel', 'Kill Cancel')}
                                                             icon={<Icon name="stop" color="red" />}
-                                                            name={ExecutionUtils.KILL_CANCEL_EXECUTION}
+                                                            name={ExecutionUtils.KILL_CANCEL_ACTION}
                                                             onClick={() =>
                                                                 handleCancelExecution(
                                                                     item,
-                                                                    ExecutionUtils.KILL_CANCEL_EXECUTION
+                                                                    ExecutionUtils.KILL_CANCEL_ACTION
```
I guess there was a bug because there is no `ExecutionUtils.KILL_CANCEL_EXECUTION` but there is `ExecutionUtils.KILL_CANCEL_ACTION`, and all orher `ExecutionUtils.{somethis}` in this file has `_ACTION` suffix.

leon-good-life commented on app/actions/manager/maintenance.ts:
```diff
@@ -9,12 +9,17 @@ import type { CancelAction, Execution } from '../../utils/shared/ExecutionUtils'
+    Required<
+        Pick<
+            Execution,
+            'id' | 'workflow_id' | 'status' | 'status_display' | 'blueprint_id' | 'deployment_id' | 'is_system_workflow'
```
I added here `| 'is_system_workflow'` because in `app/components/shared/MaintenanceModeModal.tsx` we have the following line: `<Checkmark value={item.is_system_workflow} />`

A related additional change is in `app/utils/shared/ExecutionUtils.ts` I added `is_system_workflow?: boolean;` to:
```typescript
export interface Execution {
    /* eslint-disable camelcase */
    id?: string;
    blueprint_id?: string;
    deployment_id?: string;
    status_display?: string;
    status?: string;
    scheduled_for?: string;
     workflow_id?: string;
     finished_operations?: number;
     total_operations?: number;
     is_system_workflow?: boolean; // <-- I added this line. It is in this PR. can be seen in the diff of `app/utils/shared/ExecutionUtils.ts`
     /* eslint-enable camelcase  */
 }
```

kubama commented:
> I suggest also to do the ongoing refactor of moving code from `containers` to `components` in a separate PR.

I'm not sure if such split makes much sense. In this PR you are introducing props interface but it is going to be dropped when container is merged with component.

leon-good-life commented:
I converted MaintenanceModePageMessage to function component: https://github.com/cloudify-cosmo/cloudify-stage/commit/bdbcd6615b7af7a37a3b3e7fa0dd819abcd36425
And I moved MaintenanceModePageMessage file from containers to components: https://github.com/cloudify-cosmo/cloudify-stage/commit/b8a6ce0944442f16bb0ddc6ca44af4294b5dbf2a

kubama commented on app/components/maintenance/MaintenanceModePageMessage.tsx:
```diff
@@ -1,99 +1,86 @@
+    const [maintenanceModalVisible, showMaintenanceModal, hideMaintenanceModal] = useBoolean(false);
+
+    useEffect(() => {
+        StatusPoller.getPoller()!.start();
```
Did we finally introduce the change to remove `null` from `getPoller` return type?

leon-good-life replied:
I removed it: [bd92385](https://github.com/cloudify-cosmo/cloudify-stage/pull/2245/commits/bd9238524f68d8182204ca81aae444bb4f5a265c)

kubama replied:
Can we remove `!` from here now?

leon-good-life replied:
I removed it: [8828785](https://github.com/cloudify-cosmo/cloudify-stage/pull/2245/commits/8828785a6d9db4fc4ce0ec9d7918453393df9654)

kubama commented on app/components/maintenance/MaintenanceModePageMessage.tsx:
```diff
@@ -1,99 +1,86 @@
-
-    render() {
-        SplashLoadingScreen.turnOff();
+    }, [maintenanceStatus, navigateToHome]);
```
Why is `navigateToHome` included here?

leon-good-life replied:
I added it by mistake. I made a commit that removed `navigateToHome`: [ec8fc8a](https://github.com/cloudify-cosmo/cloudify-stage/pull/2245/commits/ec8fc8a0e0f043300a032e02286577e40528e816)

kubama commented on app/components/maintenance/MaintenanceModePageMessage.tsx:
```diff
@@ -1,99 +1,86 @@
-                <MessageContainer wide>
-                    <Header as="h2">{i18n.t('maintenanceMode.header', 'Maintenance mode')}</Header>
+            <MessageContainer wide>
+                <Header as="h2">{i18n.t('maintenanceMode.header', 'Maintenance mode')}</Header>
```
We can remove the default text as part of ongoing refactors.

kubama commented on app/components/maintenance/MaintenanceModePageMessage.tsx:
```diff
@@ -1,99 +1,86 @@
+export default function MaintenanceModePageMessage() {
+    const isFetchingClusterStatus = useSelector((state: ReduxState) => state.manager.clusterStatus.isFetching);
+    const maintenanceStatus = useSelector((state: ReduxState) => state.manager.maintenance);
+    const canMaintenanceMode = useSelector((state: ReduxState) =>
```
Can we make this variable name grammatically correct?

leon-good-life replied:
I renamed it to maintenanceMode: b57f9ab880a80e5ab2cd5bceaaabe202c146a138

It is used in this line: `{maintenanceMode && <MaintenanceModeModal show={maintenanceModalVisible} onHide={hideMaintenanceModal} />}`. so I guess the new name makes sense.

kubama replied:
Looking at how the value for this variable is set it seems that `maintenanceModePermitted` would be just a little bit more descriptive.

leon-good-life replied:
I renamed it: [0af836c](https://github.com/cloudify-cosmo/cloudify-stage/pull/2245/commits/0af836cb310c2cd6f2249375f8578f8249735e4f)

kubama commented on app/components/shared/MaintenanceModeModal.tsx:
```diff
@@ -23,43 +21,65 @@ import {
 
     function stopFetchingData() {
-        if (fetchDataPromise.current) {
+        if (fetchDataPromise.current !== null) {
```
Why do we need to add `!== null`?

leon-good-life replied:
I removed it: [e677b09](https://github.com/cloudify-cosmo/cloudify-stage/pull/2245/commits/e677b09286871c1bd7914f33ef3dbddb43e20b58)

kubama commented on widgets/maintenanceModeButton/src/MaintenanceModeButton.tsx:
```diff
@@ -41,7 +41,6 @@ const MaintenanceModeButton: FunctionComponent<MaintenanceModeButtonProps> = ({
                 disabled={activating}
                 onClick={showModal}
             />
-            {/* @ts-expect-error Props inferred by TypeScript are invalid */}
```
Could you please elaborate on this?

leon-good-life replied:
I just got the following error in my console (that running `npm run devServer`) 
```
ERROR in widgets/maintenanceModeButton/src/MaintenanceModeButton.tsx:44:14
TS2578: Unused '@ts-expect-error' directive.
    42 |                 onClick={showModal}
    43 |             />
  > 44 |             {/* @ts-expect-error Props inferred by TypeScript are invalid */}
       |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    45 |             <MaintenanceModeModal show={open} onHide={hideModal} />
    46 |         </div>
    47 |     );
```
Removing this line removes the error.

leon-good-life commented:
Results of system tests:
```

[2022-12-08T18:35:49.730Z]        Spec                                              Tests  Passing  Failing  Pending  Skipped  
[2022-12-08T18:35:49.730Z]   ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
[2022-12-08T18:35:49.730Z]   │ ✔  auth_spec.ts                             00:23        4        4        -        -        - │
[2022-12-08T18:35:49.730Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.730Z]   │ ✖  change_password_spec.ts                  01:01        4        -        3        -        1 │
[2022-12-08T18:35:49.731Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.731Z]   │ ✔  cluster_status_spec.ts                   00:23        2        2        -        -        - │
[2022-12-08T18:35:49.731Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.731Z]   │ ✔  community_spec.ts                        00:43        4        4        -        -        - │
[2022-12-08T18:35:49.731Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.731Z]   │ ✔  edit_mode_spec.ts                        05:09       11       11        -        -        - │
[2022-12-08T18:35:49.731Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.731Z]   │ ✖  getting_started_spec.ts                  02:52       14       13        1        -        - │
[2022-12-08T18:35:49.731Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.732Z]   │ ✔  license_spec.ts                          01:04       10       10        -        -        - │
[2022-12-08T18:35:49.732Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.732Z]   │ ✔  login_spec.ts                            01:12       10       10        -        -        - │
[2022-12-08T18:35:49.732Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.732Z]   │ ✔  page_spec.ts                             00:55        2        2        -        -        - │
[2022-12-08T18:35:49.732Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.732Z]   │ ✖  tenants_spec.ts                          00:24        1        -        1        -        - │
[2022-12-08T18:35:49.732Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.733Z]   │ ✔  user_configuration_spec.ts               00:20        6        6        -        -        - │
[2022-12-08T18:35:49.733Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.733Z]   │ ✖  user_menu_spec.ts                        01:03        3        -        3        -        - │
[2022-12-08T18:35:49.733Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.733Z]   │ ✔  flows/admin_spec.ts                      01:06        1        1        -        -        - │
[2022-12-08T18:35:49.733Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.733Z]   │ ✖  flows/user_spec.ts                       01:54        2        1        1        -        - │
[2022-12-08T18:35:49.733Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.733Z]   │ ✔  templateManagement/page_groups_spec      00:57        5        5        -        -        - │
[2022-12-08T18:35:49.733Z]   │    .ts                                                                                         │
[2022-12-08T18:35:49.734Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.734Z]   │ ✔  templateManagement/pages_edit_spec.      01:02        1        1        -        -        - │
[2022-12-08T18:35:49.734Z]   │    ts                                                                                          │
[2022-12-08T18:35:49.734Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.734Z]   │ ✔  templateManagement/pages_preview_sp      00:17        1        1        -        -        - │
[2022-12-08T18:35:49.734Z]   │    ec.ts                                                                                       │
[2022-12-08T18:35:49.734Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.735Z]   │ ✔  templateManagement/pages_spec.ts         00:11        1        1        -        -        - │
[2022-12-08T18:35:49.735Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.735Z]   │ ✔  templateManagement/template_managem      00:22        2        2        -        -        - │
[2022-12-08T18:35:49.735Z]   │    ent_spec.ts                                                                                 │
[2022-12-08T18:35:49.735Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.735Z]   │ ✔  templateManagement/templates_spec.t      01:41        2        2        -        -        - │
[2022-12-08T18:35:49.735Z]   │    s                                                                                           │
[2022-12-08T18:35:49.735Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.735Z]   │ ✔  widgets/agents_spec.ts                   01:03        3        3        -        -        - │
[2022-12-08T18:35:49.736Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.736Z]   │ ✔  widgets/blueprint_actions_spec.ts        00:54        5        5        -        -        - │
[2022-12-08T18:35:49.736Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.736Z]   │ ✔  widgets/blueprint_catalog_spec.ts        01:57        6        6        -        -        - │
[2022-12-08T18:35:49.736Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.736Z]   │ ✔  widgets/blueprint_info_spec.ts           00:14        2        2        -        -        - │
[2022-12-08T18:35:49.736Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.737Z]   │ ✔  widgets/blueprint_num_spec.ts            00:48        3        3        -        -        - │
[2022-12-08T18:35:49.737Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.737Z]   │ ✔  widgets/blueprint_sources_spec.ts        00:21        3        3        -        -        - │
[2022-12-08T18:35:49.737Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.737Z]   │ ✔  widgets/blueprints_spec.ts               06:13       14       14        -        -        - │
[2022-12-08T18:35:49.737Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.737Z]   │ ✖  widgets/blueprints_terraform_spec.t      11:55       16       14        2        -        - │
[2022-12-08T18:35:49.737Z]   │    s                                                                                           │
[2022-12-08T18:35:49.738Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.738Z]   │ ✔  widgets/button_link_spec.ts              00:42        7        7        -        -        - │
[2022-12-08T18:35:49.738Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.738Z]   │ ✔  widgets/cloud_num_spec.ts                00:09        1        1        -        -        - │
[2022-12-08T18:35:49.738Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.738Z]   │ ✔  widgets/cluster_status_spec.ts           01:03        2        2        -        -        - │
[2022-12-08T18:35:49.738Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.738Z]   │ ✔  widgets/deployment_actions_spec.ts       01:22       11       11        -        -        - │
[2022-12-08T18:35:49.738Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.739Z]   │ ✔  widgets/deployment_actions_workflow      02:05        6        6        -        -        - │
[2022-12-08T18:35:49.739Z]   │    _parameters_spec.ts                                                                         │
[2022-12-08T18:35:49.739Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.739Z]   │ ✔  widgets/deployment_button_configura      00:41        3        3        -        -        - │
[2022-12-08T18:35:49.739Z]   │    tion_spec.ts                                                                                │
[2022-12-08T18:35:49.739Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.739Z]   │ ✔  widgets/deployment_button_deploymen      04:01       15       15        -        -        - │
[2022-12-08T18:35:49.739Z]   │    t_inputs_spec.ts                                                                            │
[2022-12-08T18:35:49.739Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.739Z]   │ ✔  widgets/deployment_button_spec.ts        01:45        9        9        -        -        - │
[2022-12-08T18:35:49.739Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.740Z]   │ ✖  widgets/deployment_info_spec.ts          00:53        4        3        1        -        - │
[2022-12-08T18:35:49.740Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.740Z]   │ ✔  widgets/deployment_num_spec.ts           01:28        7        7        -        -        - │
[2022-12-08T18:35:49.740Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.740Z]   │ ✖  widgets/deployments_spec.ts              02:22       14       12        2        -        - │
[2022-12-08T18:35:49.740Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.740Z]   │ ✖  widgets/deployments_view_spec.ts         13:02       32       31        1        -        - │
[2022-12-08T18:35:49.740Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.741Z]   │ ✔  widgets/events_spec.ts                   00:15        2        2        -        -        - │
[2022-12-08T18:35:49.741Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.741Z]   │ ✔  widgets/executions_spec.ts               00:36        8        8        -        -        - │
[2022-12-08T18:35:49.741Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.741Z]   │ ✔  widgets/executions_status_spec.ts        00:10        1        1        -        -        - │
[2022-12-08T18:35:49.741Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.741Z]   │ ✖  widgets/filter_spec.ts                   02:08        3        2        1        -        - │
[2022-12-08T18:35:49.741Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.742Z]   │ ✔  widgets/filters_spec.ts                  06:53       19       19        -        -        - │
[2022-12-08T18:35:49.742Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.742Z]   │ ✔  widgets/labels_spec.ts                   00:40        3        3        -        -        - │
[2022-12-08T18:35:49.742Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.742Z]   │ ✔  widgets/maintenance_mode_button_spe      00:48        3        3        -        -        - │
[2022-12-08T18:35:49.742Z]   │    c.ts                                                                                        │
[2022-12-08T18:35:49.742Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.742Z]   │ ✔  widgets/nodes_spec.ts                    00:25        1        1        -        -        - │
[2022-12-08T18:35:49.743Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.743Z]   │ ✔  widgets/nodes_stats_spec.ts              00:18        1        1        -        -        - │
[2022-12-08T18:35:49.743Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.743Z]   │ ✔  widgets/only_my_resources_spec.ts        00:09        1        1        -        -        - │
[2022-12-08T18:35:49.743Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.744Z]   │ ✔  widgets/outputs_capabilities_spec.t      00:31        3        3        -        -        - │
[2022-12-08T18:35:49.744Z]   │    s                                                                                           │
[2022-12-08T18:35:49.744Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.744Z]   │ ✔  widgets/plugins_catalog_spec.ts          01:23        4        4        -        -        - │
[2022-12-08T18:35:49.744Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.744Z]   │ ✔  widgets/plugins_num_spec.ts              00:34        3        3        -        -        - │
[2022-12-08T18:35:49.744Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.744Z]   │ ✔  widgets/plugins_spec.ts                  01:25        3        3        -        -        - │
[2022-12-08T18:35:49.744Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.744Z]   │ ✔  widgets/secretProviders_spec.ts          00:10        1        1        -        -        - │
[2022-12-08T18:35:49.745Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.745Z]   │ ✔  widgets/secrets_spec.ts                  00:45        1        1        -        -        - │
[2022-12-08T18:35:49.745Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.745Z]   │ ✔  widgets/servers_num_spec.ts              00:41        1        1        -        -        - │
[2022-12-08T18:35:49.745Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.745Z]   │ ✔  widgets/service_button_spec.ts           00:41        6        6        -        -        - │
[2022-12-08T18:35:49.745Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.745Z]   │ ✔  widgets/sites_map_spec.ts                00:48        4        4        -        -        - │
[2022-12-08T18:35:49.746Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.746Z]   │ ✔  widgets/sites_spec.ts                    00:57       12       12        -        -        - │
[2022-12-08T18:35:49.746Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.746Z]   │ ✔  widgets/snapshots_spec.ts                00:53        2        2        -        -        - │
[2022-12-08T18:35:49.746Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.746Z]   │ ✔  widgets/spire_manager_spec.ts            00:33        7        7        -        -        - │
[2022-12-08T18:35:49.746Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.747Z]   │ ✔  widgets/tenants_spec.ts                  01:01        1        1        -        -        - │
[2022-12-08T18:35:49.747Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.747Z]   │ ✔  widgets/text_spec.ts                     00:10        1        1        -        -        - │
[2022-12-08T18:35:49.747Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.747Z]   │ ✔  widgets/tokens_spec.ts                   00:59        5        5        -        -        - │
[2022-12-08T18:35:49.747Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.747Z]   │ ✖  widgets/topology_spec.ts                 02:19        4        3        1        -        - │
[2022-12-08T18:35:49.747Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.747Z]   │ ✔  widgets/user_defined_widgets_spec.t      00:39        3        3        -        -        - │
[2022-12-08T18:35:49.747Z]   │    s                                                                                           │
[2022-12-08T18:35:49.748Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.748Z]   │ ✔  widgets/user_groups_spec.ts              01:12        3        3        -        -        - │
[2022-12-08T18:35:49.748Z]   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
[2022-12-08T18:35:49.748Z]   │ ✔  widgets/user_management_spec.ts          01:58        3        3        -        -        - │
[2022-12-08T18:35:49.748Z]   └────────────────────────────────────────────────────────────────────────────────────────────────┘
[2022-12-08T18:35:49.748Z]     ✖  11 of 69 failed (16%)                  1:46:32      358      340       17        -        1  
```

leon-good-life commented:
Latest tests result:
```
        Spec                                              Tests  Passing  Failing  Pending  Skipped  
   ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
   │ ✔  auth_spec.ts                             00:21        4        4        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✖  change_password_spec.ts                  01:00        4        -        3        -        1 │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  cluster_status_spec.ts                   00:22        2        2        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  community_spec.ts                        00:40        4        4        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  edit_mode_spec.ts                        04:56       11       11        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✖  getting_started_spec.ts                  02:43       14       12        2        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  license_spec.ts                          01:02       10       10        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  login_spec.ts                            01:13       10       10        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  page_spec.ts                             00:52        2        2        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✖  tenants_spec.ts                          00:24        1        -        1        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  user_configuration_spec.ts               00:20        6        6        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✖  user_menu_spec.ts                        01:03        3        -        3        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  flows/admin_spec.ts                      01:07        1        1        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✖  flows/user_spec.ts                       03:09        2        1        1        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  templateManagement/page_groups_spec      00:53        5        5        -        -        - │
   │    .ts                                                                                         │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  templateManagement/pages_edit_spec.      01:00        1        1        -        -        - │
   │    ts                                                                                          │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  templateManagement/pages_preview_sp      00:16        1        1        -        -        - │
   │    ec.ts                                                                                       │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  templateManagement/pages_spec.ts         00:10        1        1        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  templateManagement/template_managem      00:21        2        2        -        -        - │
   │    ent_spec.ts                                                                                 │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  templateManagement/templates_spec.t      01:39        2        2        -        -        - │
   │    s                                                                                           │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/agents_spec.ts                   01:02        3        3        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/blueprint_actions_spec.ts        00:51        5        5        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/blueprint_catalog_spec.ts        01:57        6        6        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/blueprint_info_spec.ts           00:14        2        2        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/blueprint_num_spec.ts            00:45        3        3        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/blueprint_sources_spec.ts        00:21        3        3        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/blueprints_spec.ts               05:53       14       14        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✖  widgets/blueprints_terraform_spec.t      10:49       16       15        1        -        - │
   │    s                                                                                           │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/button_link_spec.ts              00:41        7        7        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/cloud_num_spec.ts                00:09        1        1        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/cluster_status_spec.ts           01:03        2        2        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/deployment_actions_spec.ts       01:19       11       11        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/deployment_actions_workflow      01:58        6        6        -        -        - │
   │    _parameters_spec.ts                                                                         │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/deployment_button_configura      00:39        3        3        -        -        - │
   │    tion_spec.ts                                                                                │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/deployment_button_deploymen      03:47       15       15        -        -        - │
   │    t_inputs_spec.ts                                                                            │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/deployment_button_spec.ts        01:45        9        9        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✖  widgets/deployment_info_spec.ts          00:51        4        3        1        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/deployment_num_spec.ts           01:24        7        7        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✖  widgets/deployments_spec.ts              02:21       14       13        1        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✖  widgets/deployments_view_spec.ts         12:13       32       31        1        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/events_spec.ts                   00:14        2        2        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/executions_spec.ts               00:38        8        8        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/executions_status_spec.ts        00:09        1        1        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✖  widgets/filter_spec.ts                   02:00        3        2        1        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/filters_spec.ts                  06:37       19       19        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/labels_spec.ts                   00:37        3        3        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/maintenance_mode_button_spe      00:42        3        3        -        -        - │
   │    c.ts                                                                                        │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/nodes_spec.ts                    00:24        1        1        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/nodes_stats_spec.ts              00:17        1        1        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/only_my_resources_spec.ts        00:10        1        1        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/outputs_capabilities_spec.t      00:31        3        3        -        -        - │
   │    s                                                                                           │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/plugins_catalog_spec.ts          01:12        4        4        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/plugins_num_spec.ts              00:30        3        3        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/plugins_spec.ts                  01:16        3        3        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/secretProviders_spec.ts          00:15        1        1        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/secrets_spec.ts                  00:39        1        1        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/servers_num_spec.ts              00:40        1        1        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/service_button_spec.ts           00:40        6        6        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/sites_map_spec.ts                00:47        4        4        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/sites_spec.ts                    00:58       12       12        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/snapshots_spec.ts                00:58        2        2        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/spire_manager_spec.ts            00:33        7        7        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/tenants_spec.ts                  01:01        1        1        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/text_spec.ts                     00:11        1        1        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/tokens_spec.ts                   00:59        5        5        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/topology_spec.ts                 02:14        4        4        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/user_defined_widgets_spec.t      00:38        3        3        -        -        - │
   │    s                                                                                           │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/user_groups_spec.ts              01:15        3        3        -        -        - │
   ├────────────────────────────────────────────────────────────────────────────────────────────────┤
   │ ✔  widgets/user_management_spec.ts          02:06        3        3        -        -        - │
   └────────────────────────────────────────────────────────────────────────────────────────────────┘
     ✖  10 of 69 failed (14%)                  1:43:22      358      342       15        -        1  
```

leon-good-life commented:
I started system tests on Jenkins:
https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/4093/pipeline
